### DISCUSSION TITLE
IKEv1: ignore expired client certificates?

### DISCUSSION THREAD

**Discussion by dirk077** [strongswan/strongswan#514] [Q&A]
Hi,

we're running strongswan-5.3.5-4 using the following config:

conn clients
   keyexchange=ikev1
   xauth=server
   authby=xauthrsasig
   type=tunnel
   left=a.b.c.d
   leftsubnet=0.0.0.0/0
   leftcert=valid.crt
   right=%any
   rightsourceip=%radius
   rightcert=expired.clientCert.pem,valid.clientCert.pem
   dpdaction=clear
   fragmentation=yes
   auto=add

Is there a way to allow client using the expired.clientCert.pem to connect? I hoped for rightsendcert=never, but this seems to be a IKEv2 option? 

Thanx for any hint.

**Comment by tobiasbrunner** [comment]
> Is there a way to allow client using the expired.clientCert.pem to connect? 

Not by configuring an expired certificate in `rightcert`. That's currently only possible for self-signed certificates, which are accepted if configured even if they are expired.

However, if you are able to generate a valid certificate for the client and configure that in `rightcert`, it will work even if the client sends its expired certificate.

> I hoped for rightsendcert=never, but this seems to be a IKEv2 option?

It's not, but it also can't actually influence what the other peer does. What the setting does is prevent the sending of certificate requests. Some clients might then not send their certificate (e.g. strongSwan with its default setting for `leftsendcert=ifasked`), but there is definitely no guarantee.
